src/Scripts/Ecs/Extension/EntityFactory.gd

```diff
@@ -0,0 +1,29 @@
+const Entity = preload("res://Scripts/Ecs/Core/Entity.gd")
+
+const AttackComponent = preload("res://Scripts/Ecs/Components/AttackComponent.gd")
+const CameraFollowComponent = preload("res://Scripts/Ecs/Components/CameraFollowComponent.gd")
+const ChasePlayerComponent = preload("res://Scripts/Ecs/Components/ChasePlayerComponent.gd")
+const HealthComponent = preload("res://Scripts/Ecs/Components/HealthComponent.gd")
+const PlayerMovementComponent = preload("res://Scripts/Ecs/Components/PlayerMovementComponent.gd")
+const SightComponent = preload("res://Scripts/Ecs/Components/SightComponent.gd")
+const SpriteComponent = preload("res://Scripts/Ecs/Components/SpriteComponent.gd")
+
+func create_player(tile):
+    return Entity.new(tile.x, tile.y) \
+		.add("SpriteComponent", SpriteComponent.new("Creatures", "Player")) \
+		.add("PlayerMovementComponent", PlayerMovementComponent.new()) \
+		.add("CameraFollowComponent", CameraFollowComponent.new()) \
+		.add("AttackComponent", AttackComponent.new(Constants.PLAYER_ATTACK_DAMAGE)) \
+		.add("HealthComponent", HealthComponent.new(Constants.PLAYER_MAX_HEALTH)) \
+        .add("SightComponent", SightComponent.new(Constants.PLAYER_SIGHT))
+
+func create_monster(tile):
```

Comment:
Please change these methods to take in `(x, y)` instead of tile. Tile doesn't make sense as something the entity factory should know about.